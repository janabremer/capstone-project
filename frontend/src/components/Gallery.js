import ProductPage from '../pages/ProductPage'
import ProductInfo from '../components/ProductInfo'
import styled from 'styled-components/macro'
import useAllProducts from '../hooks/useAllProducts'

export default function Gallery() {
    const { products } = useAllProducts()
    switch (products.state) {
        case 'LOADING':
            return <ProductInfo apiState={products.state} />
        case 'ERROR':
            return <ProductInfo apiState={products.state} errorMessage={products.error} />
        default:
            return(
                <GalleryStyled>
                    {products.data.map(product =>
                        <ProductPage key={product.id} productId={product.id} />
                    )}
                </GalleryStyled>
            )
    }
}

const GalleryStyled = styled.section`
    display: flex;
    flex-wrap: nowrap;
    overflow-x: scroll;
    scroll-snap-type: x mandatory;
`