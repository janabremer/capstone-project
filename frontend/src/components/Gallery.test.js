import { render } from '@testing-library/react'
import 'jest-styled-components'
import Gallery from './Gallery'

describe('Gallery', () => {
    it('renders correctly', () => {
        const { container } = render(<Gallery />)
        expect(container.firstChild).toMatchSnapshot()
    })
})