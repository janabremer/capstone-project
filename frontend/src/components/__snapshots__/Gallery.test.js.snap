// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders correctly 1`] = `
.c0 {
  background-color: var(--primary-light-transparent);
  border-radius: 25px;
  box-shadow: 0px 0px 20px var(--shadow-dark);
  display: grid;
  gap: var(--gap-medium);
  padding: 25px;
  place-items: center;
  width: 100%;
}

.c1 {
  font-size: 130%;
}

<section
  class="c0"
>
  <span
    class="c1"
  >
    loading...
  </span>
</section>
`;
